packages/sdk/src/modules/FundOperations.ts

```diff
@@ -76,7 +84,10 @@ export function withFundOperations<TBase extends MidasBaseConstructor>(Base: TBa
       amount: BigNumber
     ) {
       await this.approve(cTokenAddress, underlyingTokenAddress, amount);
-      await this.enterMarkets(cTokenAddress, comptrollerAddress, enableAsCollateral);
+      if (enableAsCollateral) {
```

Comment:
Same with this. Can we move this into the UI? As it's a field set by the UI itself anyways.
`enableAsCollateral`

So only call `enterMarket` if `enableAsCollateral` is set. 

Same with `approve` only call approve if not enough already approved. 

The functions themselves should not check if they should do it or not. Cleaning these up and they will always return a `Promise<Transaction>` and not sometimes null or such stuff.

Reply:
Makes sense! I will do like that